File: backend/src/main/java/com/codeloom/backend/dao/problem/ProblemRepositoryCustom.java
package com.codeloom.backend.dao.problem;

import com.codeloom.backend.dto.ProblemFilters;
import com.codeloom.backend.dto.ProblemListDto;
import java.util.List;

public interface ProblemRepositoryCustom {
    List<ProblemListDto> findProblemListDtos(ProblemFilters filters);
}

File: backend/src/main/kotlin/com/codeloom/backend/dao/problem/ProblemRepositoryCustom.kt
package com.codeloom.backend.dao.problem

import com.codeloom.backend.dto.ProblemFilters
import com.codeloom.backend.dto.ProblemListDto

interface ProblemRepositoryCustom {
    fun findProblemListDtos(filters: ProblemFilters): List<ProblemListDto>
}
